### PR TITLE
feat: add file uploads to request body

### PR DESCRIPTION
**Pull request by marclave**
continuation of https://github.com/scalar/scalar/pull/1966

needs some polish from @cameronrohani :)

image
image

### CODE REVIEW

File: packages/client-app/src/components/DataTable/DataTableCell.vue [hunk 1]
```diff
@@ -11,7 +11,7 @@ withDefaults(
 <template>
   <component
     :is="is"
-    class="min-h-[31px] min-w-0 border-b-[.5px] border-r-[.5px] border-r-[inherit] text-sm last:border-r-0 group-last:border-b-transparent"
+    class="min-h-[31px] h-[31px] min-w-0 border-b-[.5px] border-r-[.5px] border-r-[inherit] text-sm last:border-r-0 group-last:border-b-transparent"
```
Comment (amritk) [inline packages/client-app/src/components/DataTable/DataTableCell.vue:14]: I strongly suggest we don't use pixel values!!! Its an anti-pattern

Comment (amritk) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: just snap to the grid 😄

File: packages/client-app/src/views/Request/RequestSection/RequestBody.vue [hunk 2]
```diff
@@ -228,15 +314,28 @@ const handleFileUpload = (idx: number) => {
...
-                <span>Upload File</span>
-                <ScalarIcon
-                  class="ml-1"
-                  icon="Upload"
-                  size="xs" />
-              </ScalarButton>
+              <template v-if="activeExample?.body.binary">
+                <span class="text-c-2">{{
+                  activeExample?.body.binary.name
+                }}</span>
```
Comment (amritk) [inline packages/client-app/src/views/Request/RequestSection/RequestBody.vue:320]: nit:
```suggestion
                <span class="text-c-2">
                  {{  activeExample?.body.binary.name  }}
                </span>
```

**Review by amritk — APPROVED** [review]
Left comment about the px thing, rest looks good

File: packages/client-app/src/components/DataTable/DataTableCell.vue (hunk 1 above)
Comment (marclave) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: @antlio added this 👀 can we add a lint rule for this?

Comment (amritk) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: Well there are rare occasions where it's ok. But like 99% if the time we shouldn't. Just need to get all the devs on board!

If we are doing it a lot then our config is wrong and it should be added there.

**Comment by marclave** [comment]
worked with @geoffgscott on the binary file upload option, we will fast follow with removing axios here and using fetch : )

File: packages/client-app/src/components/DataTable/DataTableCell.vue (hunk 1 above)
Comment (antlio) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: not disagreeing but snapping the grid wouldnt maintain height consistency here.

i'll give a global eye on this and take another path to reach both 👍

Comment (amritk) [reply packages/client-app/src/components/DataTable/DataTableCell.vue:14]: Then we adjust some other spots. If everything is lined up then you don't need use any pixel values

**Pull request merged**